### DISCUSSION TITLE
System.Threading.AccessControl must be explicitly installed else exception

### DISCUSSION THREAD

**Discussion by invertedtomato** [JasperFx/marten#2631] [Q&A]
If the `System.Threading.AccessControl` NuGet package isn't explicitly installed, the following exception occurs a bunch of times at startup:

```
System.IO.FileNotFoundException: Could not load file or assembly 'System.Threading.AccessControl, Version=0.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'. The system cannot find the file specified.

File name: 'System.Threading.AccessControl, Version=0.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'
    at System.Reflection.RuntimeAssembly.InternalLoad(AssemblyName assemblyName, StackCrawlMark& stackMark, AssemblyLoadContext assemblyLoadContext, RuntimeAssembly requestingAssembly, Boolean throwOnFileNotFound)
    at System.Reflection.Assembly.Load(AssemblyName assemblyRef)
    at JasperFx.RuntimeCompiler.AssemblyGenerator.ReferenceAssembly(Assembly assembly)
```

It doesn't _appear_ to be fatal.

**Comment by gfoidl** [comment]
On which runtime / target do you build / run?

**Comment by invertedtomato** [comment]
This was building for .NET7 and running on mac-arm64. Is that what you mean?

**Comment by cocowalla** [comment]
I've had the same issue too, but on .NET 5.0 and 6.0, and for win-x64, win-x86 and linux-x64. The issue appeared with Marten v4, and I still have a package ref for `System.Threading.AccessControl` in all my projects that use Marten, though I should check and see if I actually still need it. IIRC, the stack trace revealed it was something related to Marten/Lamar's codegen that needed it.

See also previous [discussion](https://github.com/JasperFx/marten/discussions/2188) and [issue](https://github.com/JasperFx/marten/issues/1688).
